### PR TITLE
[Messenger][DX] Validate that the message exists to be able to configure its routing

### PR DESCRIPTION
**Pull request by sroze**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | ø
| License       | MIT
| Doc PR        | ø

It attempted to me already: mispelling the message name and not understanding why it didn't work... An exception letting me know that the configured message does not exists will be 👌

### CODE REVIEW

**Review by ogizanagi — APPROVED** [review]
Tests failures prove this works. But tests failures are prohibited 😄

**Review by chalasr — APPROVED** [review]
would be good to keep one test with failure expected though :)

**Comment by sroze** [comment]
> would be good to keep one test with failure expected though

I don't think it's worth it: it would means another set of yml/xml/php tests, just for these `_exists` checks :)

**Comment by ogizanagi** [comment]
> the connection in the train is so flacky

You should try taking the bus instead to send your messages more efficiently.

Alright, I'm out... -> []

**Comment by sroze** [comment]
London -> Paris in bus is a nightmare (at takes ages haha).

File: src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php [hunk 1]
```diff
@@ -1464,6 +1464,10 @@ private function registerMessengerConfiguration(array $config, ContainerBuilder
 
         $messageToSenderIdsMapping = array();
         foreach ($config['routing'] as $message => $messageConfiguration) {
+            if (!class_exists($message) && !interface_exists($message)) {
```
Comment (nicolas-grekas) [inline src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php]: missing false as 2nd arg to interface_exists

File: src/Symfony/Bundle/FrameworkBundle/Tests/DependencyInjection/Fixtures/php/messenger.php [hunk 2]
```diff
@@ -3,8 +3,8 @@
 $container->loadFromExtension('framework', array(
     'messenger' => array(
         'routing' => array(
-            'App\Bar' => array('sender.bar', 'sender.biz'),
-            'App\Foo' => 'sender.foo',
+            'Symfony\Bundle\FrameworkBundle\Tests\Fixtures\Messenger\FooMessage' => array('sender.bar', 'sender.biz'),
```
Comment (nicolas-grekas) [inline src/Symfony/Bundle/FrameworkBundle/Tests/DependencyInjection/Fixtures/php/messenger.php]: FooMessage::class

File: src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php (hunk 1 above)
Comment (sroze) [reply src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php]: Why wouldn't I want the autoload for the interface? 🤔

Comment (chalasr) [reply src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php]: autoload has been triggered by the first check already, no need to do it twice. 
Actually both calls could take `false`, there is already an `interface_exists` call l1444 above

Comment (nicolas-grekas) [reply src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php]: > autoload has been triggered by the first check already, no need to do it twice.

that's the reason indeed

> there is already an interface_exists call l1444 above

but for another interface, so not sure what you mean :)

Comment (chalasr) [reply src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php]: nvm :) commented too fast

Comment (sroze) [reply src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php]: Alright, very nice. Thanks for that, updated 👍

**Review by Tobion — APPROVED** [review]

**Comment by Tobion** [comment]
Thanks @sroze.

**Pull request merged**